src/web_task.cpp

```diff
@@ -41,8 +46,45 @@ void onWSEvent(AsyncWebSocket *server,
         Serial.printf("WebSocket client #%u disconnected\n", client->id());
         break;
     case WS_EVT_DATA:
-        Serial.printf("Raw json Message %.*s\n", len, data);
-        deserializeJson(input, data, len);
+    {
+        AwsFrameInfo * info = (AwsFrameInfo*)arg;
+        if(info->final && info->index == 0 && info->len == len)
+        {
+            // the whole message is in a single frame and we got all of it's data
+            Serial.printf("Raw json Message: %.*s\n", len, data);
+            deserializeJson(input, data, len);
+        }
+        else
+        {
+            // message is comprised of multiple frames or the frame is split into multiple packets
+			if (socketData == NULL) {
+				// allocate memory for buffer on first call
+				socketData  = (char *) malloc (SOCKET_DATA_SIZE);
```

Comment:
allocated memory is not freed any more. memory leak?!

Reply:
ahhh. i think i got the way this works. you are allocating memory only the first time a multi part message is received. future multi-part messages always use the same buffer without allocating new memory.

Reply:
Yes. This is the way it works. It would be cleaner to free it in some shutdown routine, but I am just learning C++ and ESP32, so I don't know where to put it. It should not be a problem in practise as the program never ends.

Reply:
it is not about the never ending program, but memory-leaks occur if memory is not freed but new memory is allocated. however, due to the check for NULL in this case memeory is allocated only once. but, in my experience errors in dynamic memory allocation errors are quite hard to trace. especially in embedded applications. it might be that the application works for several days but after some days the error manifests. usually it is more stable to use static memory allocation. that's what i also would suggest here. just allocate SOCKET_DATA_SIZE statically.